src/me/drton/jmavsim/UDPMavLinkPort.java

```diff
@@ -78,21 +78,25 @@ private static InetAddress getMyHostIPAddress() throws UnknownHostException, Soc
         // Look over all the network interfaces for the appropriate interface.
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
         while (networkInterfaces.hasMoreElements()) {
-           NetworkInterface iface = networkInterfaces.nextElement();
-           Enumeration<InetAddress> addresses = iface.getInetAddresses();
-           while (addresses.hasMoreElements()) {
-               InetAddress address = addresses.nextElement();
-               if (!address.isLoopbackAddress()) {
-                   if (address.isSiteLocalAddress()) {
-                       // probably a good one!
-                       return address;
-                   } else {
-                       // Found a non-loopback address that isn't site local.
-                       // Might be link local (private network), but probably better than nothing.
-                       possibleAddress = address;
-                   }
-               }
-           }
+            NetworkInterface iface = networkInterfaces.nextElement();
+            Enumeration<InetAddress> addresses = iface.getInetAddresses();
+            while (addresses.hasMoreElements()) {
+                InetAddress address = addresses.nextElement();
+                if (!address.isLoopbackAddress()) {
+                    if (address.isSiteLocalAddress()) {
+                        // probably a good one!
+                        return address;
+                    } else {
+                        // Found a non-loopback address that isn't site local.
+                        // Might be link local (private network), but probably better than nothing.
+                        possibleAddress = address;
+                        if(address instanceof Inet4Address) {
+                            possibleAddress = address;
+                            break;
```

Comment:
@LorenzMeier could you clarify what you mean by "alters in the loop"? I am assuming it was a typo of "alerts" but could you please confirm?